### PR TITLE
Make Meka-Tool and Armor lava proof

### PR DESCRIPTION
**Pull request by ToMe25**
## Changes proposed in this pull request:
 - Make Meka-Tool and Mekasuit fire & lava proof
They make you immune to fire, so why should they burn?

### CODE REVIEW

File: src/main/java/mekanism/common/registration/impl/ItemDeferredRegister.java [hunk 1]
```diff
@@ -54,6 +54,10 @@ public ITextComponent getDisplayName(@Nonnull ItemStack stack) {
         return register(name, () -> sup.apply(getMekBaseProperties()));
     }
 
+    public <ITEM extends Item> ItemRegistryObject<ITEM> registerUnburnable(String name, Function<Item.Properties, ITEM> sup) {
```
Comment (pupnewfster) [inline src/main/java/mekanism/common/registration/impl/ItemDeferredRegister.java:57]: The registerUnburnable method that already exists should probably be adjusted to just being `return register(name, Item::new);`

Comment (ToMe25) [reply src/main/java/mekanism/common/registration/impl/ItemDeferredRegister.java:57]: do you mean `return registerUnburnable(name, Item::new);` or did i miss something?

Comment (pupnewfster) [reply src/main/java/mekanism/common/registration/impl/ItemDeferredRegister.java:57]: Yes, sorry I copy pasted and only adjusted the supplier -> method reference forgot to update the method it was calling.

Comment (ToMe25) [reply src/main/java/mekanism/common/registration/impl/ItemDeferredRegister.java:57]: done.

**Pull request merged**